### ISSUE TITLE
[iOS] adjustsFontSizeToFit different behaviour on new architecture

### ISSUE THREAD

**Issue by SamuelScheit** [facebook/react-native#42044]
### Description

The new architecture behaves differently for Text components with `adjustsFontSizeToFit` in comparison to the old architecture.
The old architecture text is capped at `15px` (but would shrink, if the space wouldn't fit).
The new architecture isn't capped and the text grows as big as it fits.
The bug doesn't occur on android.

### Steps to reproduce

Run the following react-native app on the old and new architecture
```js
function App() {
  return (
    <SafeAreaView>
      <Text
        style={{fontSize: 15}}
        minimumFontScale={1}
        maxFontSizeMultiplier={1}>
        without adjustsFontSizeToFit
      </Text>
      <Text
        style={{fontSize: 15}}
        adjustsFontSizeToFit
        minimumFontScale={0.5}
        maxFontSizeMultiplier={1.5}>
        min 0.5, max: 1.5
      </Text>
      <Text
        style={{fontSize: 15}}
        adjustsFontSizeToFit
        minimumFontScale={1}
        maxFontSizeMultiplier={1}>
        min 1, max: 1
      </Text>
    </SafeAreaView>
  );
}
```

### React Native Version

0.73.1

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.2
  CPU: (20) arm64 Apple M1 Ultra
  Memory: 405.34 MB / 64.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.17.1
    path: ~/.nvm/versions/node/v18.17.1/bin/node
  Yarn:
    version: 1.22.19
    path: ~/.nvm/versions/node/v18.17.1/bin/yarn
  npm:
    version: 9.8.1
    path: ~/.nvm/versions/node/v18.17.1/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.12.1
    path: /opt/homebrew/opt/ruby/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.0
      - iOS 17.0
      - macOS 14.0
      - tvOS 17.0
      - visionOS 1.0
      - watchOS 10.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10671973
  Xcode:
    version: 15.1/15C5028h
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 20.0.2
    path: /Library/Java/JavaVirtualMachines/jdk-20.jdk/Contents/Home/bin/javac
  Ruby:
    version: 3.2.2
    path: /opt/homebrew/opt/ruby/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.1
    wanted: 0.73.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
-
```

### Reproducer

https://snack.expo.dev/2qB26Vtg9

## iOS
left old architecture | right new architecture

<div>
<img width="200" src="https://github.com/facebook/react-native/assets/34555296/d02441ab-ea9e-4bdf-a696-d3937ae22c8a" />
<img width="200" src="https://github.com/facebook/react-native/assets/34555296/f192d1db-c2e9-4b4d-9ff2-51abea94e0f4" />
</div>

## Android (is Working correctly)

left old architecture | right new architecture

<div>
<img width="200" src="https://github.com/facebook/react-native/assets/34555296/03e92c83-d9d4-4392-b1ea-3f30ac8cc419" />
<img width="200" src="https://github.com/facebook/react-native/assets/34555296/c8637493-8a7b-451b-a60b-9041b0937796" />
</div>

**Comment by fabriziobertoglio1987** [comment]
Do you still experience this issue? If yes, I will publish the fix in the react-native-improved package https://github.com/fabriziobertoglio1987/react-native-improved. Thanks a lot

**Comment by cortinico** [comment]
Related to #43104

**Comment by cortinico** [comment]
Fixed by https://github.com/facebook/react-native/pull/44075

**Comment by cortinico** [comment]
> Fixed by #44075

Actually not as this is iOS and not Android
